master/buildbot/reporters/github.py

```diff
@@ -141,6 +141,8 @@ def send(self, build):
             repoOwner, repoName = project.split('/')
         else:
             giturl = giturlparse(sourcestamps[0]['repository'])
+            if not giturl:
```

Comment:
Well, it is not normal that sourcestamp is not set on rebuild. github reporter should work on rebuild.

It does work on nine.buildbot.net(I often rebuild, and it updates the status on github), so I think there is an issue with your configuration.

Reply:
ok, I will investigate this more. Might just be a side effect of #3683 even ( the last few changes to github reports/hooks seems to have broken GHE a lot )